### PR TITLE
V2 auth is not working for some servers with project.json restore

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/NuGet/Home/issues/1082

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @emgarten \ @deepakaravindr

File: src/NuGet.Protocol.Core.v3/RemoteRepositories/HttpSource.cs [hunk 1]
```diff
@@ -53,97 +53,123 @@ internal Task<HttpSourceResult> GetAsync(string uri, string cacheKey, TimeSpan c
             Logger.LogVerbose(string.Format(CultureInfo.InvariantCulture, "  {0} {1}.", "GET", uri));
 
             ICredentials credentials = CredentialStore.Instance.GetCredentials(_baseUri);
-retryWithAuthentication:
```
Comment (deepakaravindr) [inline src/NuGet.Protocol.Core.v3/RemoteRepositories/HttpSource.cs:56]: This was checked-in. Seriously?! :)

File: src/NuGet.Protocol.Core.v3/RemoteRepositories/STSAuthHelper.cs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+#if !DNXCORE50
+using System.ServiceModel;
+using System.ServiceModel.Security;
+#endif
+using System.Text;
+using NuGet.Configuration;
+
+namespace NuGet.Protocol.Core.v3
+{
+    public class STSAuthHelper
```
Comment (deepakaravindr) [inline src/NuGet.Protocol.Core.v3/RemoteRepositories/STSAuthHelper.cs:22]: Looks similar to http://nuget.codeplex.com/SourceControl/latest#src/Core/Http/STSAuthHelper.cs, but not the same. Why?!

File: src/NuGet.Protocol.Core.v3/RemoteRepositories/STSAuthHelper.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    {
+        /// <summary>
+        /// Response header that specifies the WSTrust13 Windows Transport endpoint.
+        /// </summary>
+        /// <remarks>
+        /// TODO: Is there a way to discover this \ negotiate this endpoint?
```
Comment (deepakaravindr) [inline src/NuGet.Protocol.Core.v3/RemoteRepositories/STSAuthHelper.cs:28]: Please file a bug and remove the TODO

File: src/NuGet.Protocol.Core.v3/RemoteRepositories/STSAuthHelper.cs (hunk 2 above)
Comment (pranavkm) [inline src/NuGet.Protocol.Core.v3/RemoteRepositories/STSAuthHelper.cs:22]: It's ported over.

Comment (deepakaravindr) [inline src/NuGet.Protocol.Core.v3/RemoteRepositories/STSAuthHelper.cs:22]: Is it because we do not use WebRequest anymore?!

Comment (pranavkm) [inline src/NuGet.Protocol.Core.v3/RemoteRepositories/STSAuthHelper.cs:22]: That plus this assembly does not depend on NuGet.Core.

**Comment by pranavkm** [comment]
Does this look good besides the TODO?

**Comment by deepakaravindr** [comment]
Looks good to me

**Comment by emgarten** [comment]
:shipit:

**Pull request closed without merging**